lib/ravendb/documents/session/in_memory_document_session_operations.rb

```diff
@@ -0,0 +1,130 @@
+module RavenDB
+  class InMemoryDocumentSessionOperations
+    attr_reader :included_documents_by_id
+
+    def initialize
+      @_known_missing_ids = {}
+      @included_documents_by_id = {}
+      @documents_by_id = {}
+    end
+
```

Comment:
In Java there is a number of methods related to session events. https://github.com/ravendb/ravendb-jvm-client/blob/48d6516935eb78009e000b6f72687318d7fb088f/src/main/java/net/ravendb/client/documents/session/InMemoryDocumentSessionOperations.java#L81

In JS we use extend built-in `EventEmitter` class with `on(eventName, handler)` and `removeEventListener(eventName, handler)` methods. Is there a Ruby built-in we can use for this purpose, or should we go the Java way?

Reply:
Oh I see it extends `Observable`. That's fine.